Source/engine.cpp

```diff
@@ -36,6 +36,52 @@ const uint32_t RndInc = 1;
  */
 const uint32_t RndMult = 0x015A4E35;
 
+namespace {
+
+template <bool SkipColorIndexZero>
+void BufferBlit(const CelOutputBuffer &src, SDL_Rect srcRect, const CelOutputBuffer &dst, Point dstPosition)
+{
+	// We do not use `SDL_BlitSurface` here because the palettes may be different objects

```

Comment:
Would it make sens to check if they use the same palette and then use SDL?

Reply:
Maybe as an optimization but I think this is unlikely to make much difference. We'd also need to fiddle around with COLOR_KEY for the SkipColorIndexZero version, and that's a bit trickier.